src/Globals.cpp

```diff
@@ -1,14 +1,47 @@
 #include "Globals.h"
+
 #include "Constants.h"
 #include "world_interface/data.h"
 
-#include <vector>
 #include <atomic>
+#include <vector>
+
+using robot::types::motorid_t;
+using namespace Constants::arm;
+
+namespace {
+
+navtypes::Vectord<IK_MOTORS.size()> getSegLens() {
+	navtypes::Vectord<IK_MOTORS.size()> ret;
+	for (std::size_t i = 0; i < IK_MOTORS.size(); i++) {
+		ret[i] = SEGMENT_LENGTHS.at(IK_MOTORS[i]);
+	}
+	return ret;
+}
+
+navtypes::Vectord<IK_MOTORS.size()> getJointLimits(bool getLow) {
+	navtypes::Vectord<IK_MOTORS.size()> ret;
+	for (std::size_t i = 0; i < IK_MOTORS.size(); i++) {
+		const auto& limits = JOINT_LIMITS.at(IK_MOTORS[i]);
+		ret[i] = getLow ? (limits.first * .001) * (M_PI / 180.0)
+						: (limits.second * .001) * (M_PI / 180.0);
```

Comment:
Convert ret[i] after setting, no need to duplicate. Also divide by 1000 instead of multiplying by 0.001.

Reply:
Actually, after the for just do `ret *= M_PI / 180.0 / 1000.0;`